### ISSUE TITLE
Report Back Transactional not necessarly in correct language

### ISSUE THREAD

**Issue by deezone** [DoSomethingArchive/legacy-website#5568]
A user report back transactional template is based on the user's location at the time of report back rather than the language of the campaign that's being reported back on (See `dosomething_mbp_get_template_name()` for generation of template name logic). This seems to currently workout but when campaigns are in several languages for global users there's a chance the report back transactional will be sent in the wrong language.

**Related**: The `campaign_language` value should also be in the payload of both the `campaign_signup` and `campaign_reportback` transactions. The Message Broker system will use these values when generating the Global Digest.

**Related / duplicate**:
- https://github.com/DoSomething/phoenix/issues/5563

cc: @mikefantini

**Comment by marahml** [comment]
@DeeZone aren't these stored as multiple templates in Mandrill? As in, shouldn't we just be triggering the correct template per country?

**Comment by deezone** [comment]
@marahml Yes, that's the idea.  But...

If a user from Brazil signs up for an English campaign they will end up getting the transactional in portuguese (`mb-user-register-BR`) rather than in the language of the campaign they signed up for. This will also be confusing as the three steps listed in the transactional will be in English while the rest of the message will be in portuguese.

This issue is about using the language of the campaign to determine what template / language to send the transactional message in rather than the current user location.

In summary, "... just be triggering the correct template per country" should perhaps be "... just be triggering the correct template per campaign language".

**Comment by blisteringherb** [comment]
@DeeZone Under what circumstances would a Brazilian user sign up for an english campaign? Also, when you say english campaign, do you mean US english or global english? I'm thinking this would only apply to BR and MX users signing up for US english campaigns. If there's a valid translation for the user's language and they hit the bare URL, they'll be redirected to the translated node, so they can never sign up for a campaign if it has a valid translation for them. In terms of BR and MX signing up for US campaigns, they'd have to explicitly visit the /us/ URL to do that signup. It seems like a bit of an edge case.

@mikefantini How do we want to address BR and MX users who sign up for english campaigns?

**Comment by mikefantini** [comment]
@blisteringherb - users should receive a campaign transactional email in the language of the campaign they signed up for.

So if I sign up for a campaign that's in GL EN, I should receive an email in GL EN, regardless of where I'm located.

If I sign up for a campaign that's in US EN, I should receive an email in US EN, again regardless of where I'm located.

**Comment by mikefantini** [comment]
@blisteringherb @DeeZone is this still an issue?
